Core/Python/update_firmware_using_catalog.py

```diff
@@ -144,7 +154,40 @@ def catalog_creation_payload(**kwargs):
         }
     }
 
-def catalog_creation(ip_address, headers, **kwargs):
+def catalog_refresh_payload(repoId):
+    refresh_payload = {
+            "CatalogIds": [repoId],
+            "AllCatalogs": "false"
+    }
+    return refresh_payload
+
+
+def create_or_refresh_catalog(ip_address, headers, **kwargs):
+    """Get all catalogs first"""
+    url = 'https://%s/api/UpdateService/Catalogs' % ip_address
+    status, data = request(url=url, header=headers, method='GET')
+    if status != 200:
+        raise Exception("Unable to get the catalog", data)
+    else:
+        allrepoProfiles  = data["value"]
+        if allrepoProfiles is not None and data["@odata.count"] != 0:
```

Comment:
I think "if allrepoProfiles" is sufficient - not if allrepoProfiles is not None

Reply:
Done